client/state/data-layer/wpcom/sites/comments/likes/new/index.js

```diff
@@ -23,13 +25,29 @@ export const likeComment = ( { dispatch }, action ) => {
 	}, action ) );
 };
 
-export const updateCommentLikes = ( { dispatch }, { siteId, postId, commentId }, next, { like_count } ) => dispatch( local( {
-	type: COMMENTS_LIKE,
-	siteId,
-	postId,
-	commentId,
-	like_count
-} ) );
+export const updateCommentLikes = ( { dispatch }, { siteId, postId, commentId, status }, next, { like_count } ) => {
+	dispatch( local( {
+		type: COMMENTS_LIKE,
+		siteId,
+		postId,
+		commentId,
+		like_count
+	} ) );
+
+	if ( isEnabled( 'comments/undo-in-datalayer' ) && 'unapproved' === status ) {
```

Comment:
Instead of adding a feature flag here, why not add notice middleware for `COMMENTS_LIKE` we can for example check the current route and only dispatch a notice if we're in the comments management path.